core/trino-main/src/main/java/io/trino/operator/project/GeneratedPageProjection.java

```diff
@@ -64,6 +66,12 @@ public InputChannels getInputChannels()
         return inputChannels;
     }
 
+    @Override
```

Comment:
Can we make `InputChannels` accept `int[] nonLazyChannels` during construction and make `InputChannels` return loaded channels automatically?

Then we wouldn't need `isChannelUnconditionallyEvaluated` method